powertools-serialization/src/main/java/software/amazon/lambda/powertools/utilities/jmespath/Base64GZipFunction.java

```diff
@@ -51,7 +51,7 @@ protected <T> T callFunction(Adapter<T> runtime, List<FunctionArgument<T>> argum
 
     public static String decompress(byte[] compressed) {
         if (compressed == null || compressed.length == 0) {
-            return "";
+            return null;
```

Comment:
If the `callFunction` is called with null it throws a `io.burt.jmespath.function.ArgumentTypeException: Invalid argument type calling "powertools_base64_gzip": expected string but was null` because of the argument constraint in the constructor

```
public Base64GZipFunction() {
        super("powertools_base64_gzip", ArgumentConstraints.typeOf(JmesPathType.STRING));
    }
```

I added a new test to catch this scenario and I followed [the instructions](https://github.com/burtcorp/jmespath-java/blob/master/README.md#configuration) of the library's documentation on how to handle cases where the function receives an argument of wrong type.

The method `decompress` on the other hand can get called separately with a null argument

Reply:
I misinterpeted your comment. I thought you were referencing to the `callFunction` being invoked with null argument.

I fixed also the part you were actually referring too now.